pages/shields/led_matrix.rst

```diff
@@ -1,4 +1,147 @@
+    tm.write([255,255,255,255,255,255,255,255])  # 255 = 0b11111111
+
+Now all LEDs on the LED Matrix should be illuminated! Now lets run through what
+we just did:
+
+- Imported the MicroPython ``machine`` module -- we need this to configure our
+  pins before passing them to our ``tm1640`` driver
+- Imported our ``wemos`` module -- we need this to get the pin information to
+  then configure the correct pins for communicating with the LED Matrix driver
+  chip
+- Imported the ``tm1640`` module -- this is the MicroPython driver that will
+  provide us easy control over the ``tm1640`` LED matrix driver chip on the
+  shield
+- Created a ``tm`` object that represents our LED Matrix driver. This took two
+  parameters, the two pins that are used for communicating with the ``tm1640``
+  chip (a clock pin and a data in/out pin)
+- Wrote an 8-long list filled with 255's to our ``tm`` object -- this is what
+  turns the LED Matrix LEDs on!
+  
+The format of the list that we wrote to the ``tm`` object is that each element
+of the list represents a row, and each bit in that element represents one of
+the LEDs in that row, with 1 being illuminated and 0 being off. As
+``255 == 0b11111111``, and we wrote 255 to all 8 elements of the list, we
+illuminated all 8 LEDs, on all 8 rows.
+
+Our driver library also gives us access to a ``brightness()`` method on our
+``tm`` object, if we wanted to reduce the brightness of the LED matrix. This
+takes a single argument of an integer from 0 to 7, where 0 is the minimum
+brightness, and 7 is the maximum (this is the default). So if we wanted to
+reduce brightness by a bit but not all the way we could do::
+
+    tm.brightness(3)
+
+If you wanted to (slightly) improve visualisation of what you are writing to
+the matrix in your code, you could format it like so::
+
+    tm.write([
+        0b00000000,
+        0b01100110,
+        0b01100110,
+        0b00000000,
+        0b00000000,
+        0b10000001,
+        0b01111110,
+        0b00000000,
+    ])
+
+The last thing we have the ability to do is write a `MicroPython frame buffer`_
+to the device (treating it like an 8x8 monochrome display). This is done like
+so::
+
+    # Instantiate our 8x8 frame buffer
+    import framebuf  # Bring in the frame buffer library
+    buf = bytearray(8)  # Reserve 8 bytes of memory for the frame buffer
+    fb = framebuf.FrameBuffer(buf, 8, 8, framebuf.MONO_HMSB)
+
+    # Draw things into our frame buffer
+    fb.text('!', 0, 0, 1)  # Draw an !
+    fb.hline(3, 7, 2, 1)  # Supplement the bottom of the ! as the font is 7x7
+    fb.vline(0, 0, 8, 1)  # Draw line down left side
+    fb.vline(7, 0, 8, 1)  # Draw line down right side
+    fb.pixel(1, 0, 1)  # Extend end of lines
+    fb.pixel(1, 7, 1)
+    fb.pixel(6, 0, 1)
+    fb.pixel(6, 7, 1)
+
+    # Draw the buffer of the frame buffer to the "display"
+    tm.write_hmsb(buf)  # Note that this takes buf, not fb
```

Comment:
Hmm, should we change the library to accept a FrameBuffer?

Reply:
Probably, time permitting - confusing otherwise